### PR TITLE
fix: 10114 MongoDB Fetch with DBRef - fix  with codec

### LINKED ISSUE #10114: [Bug]: Mongo plugin: plugin throws error if output contains DBref type
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

- If any of output documents contain a DBref type data, then the plugin throws error on action execution. 
- Currently, we use `Document::toJson()` method for converting the Mongo driver output to json format. This method is deprecated and uses default codecs only. 
- We need to replace `Document::toJson()` method with the one that used custom codes and need to define a codec to convert `DBref` type to JSON.
- Ref: https://stackoverflow.com/questions/31827635/resolve-dbref-into-json  

### Steps To Reproduce

1. Try to read a Mongo document which has `DBref` type. 

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by leotom2000**
## Description

- Reactive MongoDB driver does not support fetching DBRef directly.
- Applied Codec so that fetching happens.
- TBD: JUnit TC will be added later due to some issue with my local setup environment. Tracking it here: https://github.com/appsmithorg/appsmith/issues/12591

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

**Manual**
Input:
```
{
_id:61d2afdc27368d17882c8636
inventory:DBRef(stores, 1, testdb)
}
```
Output
 ```
 {
    "_id": "61d2afdc27368d17882c8636",
    "inventory": {
      "$db": "testdb",
      "$ref": "stores",
      "$id": 1
   }
```
**Unit Test**
Some issues found, created a issue to create unit test
[https://github.com/appsmithorg/appsmith/issues/12591](https://github.com/appsmithorg/appsmith/issues/12591)

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by sumitsum** [comment]
@leotom2000 jfyi, please keep the branch name as either `fix/my_branch_name` or `feature/my_branch_name` to keep up with the guidelines as mentioned here: https://github.com/appsmithorg/appsmith/blob/release/contributions/CodeContributionsGuidelines.md

File: app/server/appsmith-plugins/mongoPlugin/src/main/java/com/external/plugins/MongoPlugin.java [hunk 1]
```diff
@@ -332,7 +356,12 @@ public MongoPlugin(PluginWrapper wrapper) {
                     )
                     .flatMap(mongoOutput -> {
                         try {
-                            JSONObject outputJson = new JSONObject(mongoOutput.toJson());
+                            DocumentCodec documentCodec = new DocumentCodec(
```
Comment (sumitsum) [inline app/server/appsmith-plugins/mongoPlugin/src/main/java/com/external/plugins/MongoPlugin.java:359]: Please add a comment here explaining why this coded was required, along with the URL for the Github issue.

**Review by sumitsum — CHANGES_REQUESTED** [review]

File: app/server/appsmith-plugins/mongoPlugin/src/main/java/com/external/plugins/MongoPlugin.java (hunk 1 above)
Comment (leotom2000) [reply app/server/appsmith-plugins/mongoPlugin/src/main/java/com/external/plugins/MongoPlugin.java:364]: Updated code comments

**Comment by sumitsum** [comment]
Closing in favor of https://github.com/appsmithorg/appsmith/pull/12670

**Pull request closed without merging**